### PR TITLE
🔨 Replace `imgaug` with Native PyTorch Transforms

### PR DESCRIPTION
**Pull request by samet-akcay**
## 📝 Description

- Removed `imgaug` dependency from project requirements
- Replaced `Augmenter` class with new `PerlinAnomalyGenerator` using torchvision transforms
- Refactored perlin noise generation to use PyTorch native operations
- Added new `MultiRandomChoice` transform class for flexible augmentation pipelines
- Updated all dependent modules (DRAEM, DSR) to use the new implementation
- 🛠️ Fixes [# (issue number)](https://github.com/openvinotoolkit/anomalib/issues/2429)

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

### Before (using imgaug):
```python 
from imgaug import augmenters as iaa

# Define augmentations
augmenters = [
  iaa.GammaContrast((0.5, 2.0), per_channel=True),
  iaa.MultiplyAndAddToBrightness(mul=(0.8, 1.2), add=(-30, 30)),
  iaa.pillike.EnhanceSharpness(),
  # ... more imgaug augmenters
]

# Apply rotation
rot = iaa.Sequential([iaa.Affine(rotate=(-90, 90))])
perlin_noise = rot(image=perlin_noise_np)
```

### After (Using torch vision)

```python 
from torchvision.transforms import v2

# Define augmentations using MultiRandomChoice

augmenters = MultiRandomChoice(
    transforms=[
        v2.ColorJitter(contrast=(0.5, 2.0)),
        v2.Compose([
        v2.Lambda(lambda x: x torch.empty(1).uniform(0.8, 1.2).item()),
        v2.Lambda(lambda x: torch.clamp(x + torch.empty(1).uniform_(-30/255, 30/255).item(), 0, 1)),]),
        v2.RandomAdjustSharpness(sharpness_factor=2.0, p=1.0),
        # ... more torchvision transforms
    ],
    num_transforms=3,
    fixed_num_transforms=True
)

# Apply rotation
generator = PerlinAnomalyGenerator(
    anomaly_source_path="path/to/images",
    blend_factor=(0.1, 0.9),
    probability=0.5,
    rotation_range=(-90, 90)
)
augmented_image, mask = generator(image)
```

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/data/utils/generators/perlin.py [hunk 1]
```diff
@@ -1,160 +1,317 @@
...
-    noise = np.zeros(shape)
-    frequency = 1
-    amplitude = 1
-    for _ in range(octaves):
-        noise += amplitude * generate_perlin_noise_2d(shape, (frequency * res[0], frequency * res[1]))
-        frequency *= 2
-        amplitude *= persistence
-    return noise
 
+def generate_perlin_noise(
```
Comment (ashwinvaidya17) [inline src/anomalib/data/utils/generators/perlin.py:16]: Maybe for a different PR but it would be nice if we could remove inline methods.

**Review by ashwinvaidya17 — COMMENTED** [review]
One minor comment

File: tests/unit/data/validators/numpy/test_image.py [hunk 2]
```diff
@@ -185,7 +185,7 @@ def test_validate_gt_label_valid_string_input(self) -> None:
         validated_labels = self.validator.validate_gt_label(["0", "1"])
         assert isinstance(validated_labels, np.ndarray)
         assert validated_labels.dtype == bool
-        assert np.array_equal(validated_labels, np.array([False, True]))
+        assert np.array_equal(validated_labels, np.array([True, True]))
```
[the remark this answers is not in the document]
Comment (samet-akcay) [reply tests/unit/data/validators/numpy/test_image.py:188]: with numpy 2.0, yes

**Review by djdameln — APPROVED** [review]
Did you confirm that the generated anomalous images look similar, visually, as the images generated with the previous approach?

**Comment by samet-akcay** [comment]
> rated anomalous images look similar, visually, as the images generated with the previous approach?

Here is a comparison 
image

image

**Pull request merged**